src/main/java/sirius/kernel/xml/Outcall.java

```diff
@@ -110,8 +111,11 @@ public class Outcall {
      * <p>
      * These hosts are blacklisted for a short amount of time ({@link #connectTimeoutBlacklistDuration}) to prevent
      * cascading failures.
+     * <p>
+     * We log a warning for each host which is blacklisted and is called again. To make sure this doesn't spam the logs,
+     * we only log once and then set the second value of the tuple to <tt>true</tt> to prevent further log messages.
      */
-    private static final Map<String, Long> timeoutBlacklist = new ConcurrentHashMap<>();
+    private static final Map<String, Tuple<Long, Boolean>> timeoutBlacklist = new ConcurrentHashMap<>();
```

Comment:
**Optional**: a **record** would be cleaner than a Tuple and could encapsulate some helper methods to substitute the null checks done below.

Reply:
You're right, a record would be cleaner and easier to understand. I also tried a record at first, but I have to admit that - for this small task - a tuple might be enough. 

Correct me if I'm wrong, but a record only transports data that is final. Therefore, we could not call a setter, but would always have to recreate the record, or?